app/forms/mappings/Constraints.scala

```diff
@@ -81,6 +81,22 @@ trait Constraints {
         Invalid(errorKey, minimum)
     }
 
+  protected def numericAndCorrectLength(minimum: Int, maximum: Int): Constraint[String] =
+    Constraint {
+      case str if isNumeric(str) && hasCorrectLength(str, minimum, maximum)  =>
+        Valid
+      case str if isNumeric(str) && !hasCorrectLength(str, minimum, maximum) =>
+        Invalid("commodityCode.error.length")
+      case _                                                                 =>
+        Invalid("commodityCode.error.nonNumeric")
+    }
```

Comment:
```suggestion
   protected def numericAndCorrectLength(minimum: Int, maximum: Int): Constraint[String] =
    Constraint { str =>
      if (isNumeric(str)) {
        if (hasCorrectLength(str, minimum, maximum)) {
          Valid
        }
        else {
          Invalid("commodityCode.error.length")
        }
      } else {
        Invalid("commodityCode.error.nonNumeric")
      }
    }
```

Reply:
The above is just a suggestion but it removes some code duplication and reads a bit better